resources/scripts.js

```diff
@@ -125,112 +221,367 @@ function buttonPosition(loc) {
+    }
+    switch (true) {
+        case (color > 0):
+            // White
+            break;
+        case (color >= 5):
+            // Red
+            break;
+        case (color >= 25):
+            // Green
+            break;
+        case (color >= 100):
+            // Black
+            break;
+        case (color >= 500):
+            // Orange
+            break;
+    }
+    return;
 }
 
-function playerLose() {
-    displayMessage('You lose')
+function payOut(isprimary, win, amount) {
+    if (isprimary === false) {
+        if (win === true) {
+            playerMoney += (amount * 2);
+            return;
+        }
+        return;
+    }
+    if (win === true) {
+        buildSummary(PRIMARYWIN);
+        lastPlayerMoney = playerMoney;
+        playerMoney += (amount * 2);
+        pushSecondaryBets();
+        resetGame();
+        moneyChange(0);
+        return;
+    }
+    buildSummary(PRIMARYLOSE);
     if (playerMoney === 0) {
         gameOver();
         return;
     }
-    buttonPosition(1);
-    betAmount = 0;
-    rollButtonState(false);
-    betButtonState(true);
+    displayMessage(rollSummary);
+    rollSummary = null;
+    buttonPosition(playerRoll);
+    resetGame();
+    moneyChange(0);
+    return;
 }
 
-function placeBet() {
-    betAmount = betDialog.value;
+function makeBet() {
+    betAmount = betDialogElement.value;
     betAmount = parseInt(betAmount);
-    if (validateBet() === false) {
+    if (validateBet(betAmount) === false) {
         return;
     }
+    lastPlayerMoney = playerMoney;
     playerMoney -= betAmount;
-    displayMoney.innerHTML = playerMoney;
-    rollButtonState(true);
-    betButtonState(false);
+    moneyChange(betAmount);
+    buttonStates();
+    return;
+}
+
+function makeCome() {
+    let comeAmount = betDialogElement.value;
+    comeAmount = parseInt(comeAmount);
+    if (validateBet(comeAmount) === false) {
+        return;
+    }
+    lastPlayerMoney = playerMoney;
+    playerMoney -= comeAmount;
+    secondaryBetList.push(new SecondaryBet(1, comeAmount));
+    secondaryBetListO[1] = comeAmount;
+    moneyChange(comeAmount);
+    betDialogElement.disabled = true;
+    comeButtonElement.disabled = true;
+    return;
+}
+
+function makePlace(num) {
+    placeDialogAmount.setAttribute('max', playerMoney);
+    makePlaceNumClicked = num;
+    placeDialog.showModal();
+    return;
+}
+
+class SecondaryBet {
+    constructor(point, amount) {
+        this.point = point;
+        this.amount = amount;
+    }
+}
+
+function removeBet(index) {
+    if (index === 0) {
+        secondaryBetList = secondaryBetList.slice(1);
+    } else if (index === secondaryBetList.length) {
+        secondaryBetList = secondaryBetList.slice(0, -1);
+    } else {
+        const halfBefore = secondaryBetList.slice(0, index);
+        const halfAfter = secondaryBetList.slice(index + 1);
+        secondaryBetList = halfBefore.concat(halfAfter);
+    }
+    return;
+}
+
+function pushSecondaryBets() {
```

Comment:
Needs to be refactored to use object rather than array. Causes NaN when secondary bets are active and point wins.

Reply:
Function refactored though NaN issue exists elsewhere (makeCome?).